graylog2-web-interface/src/components/common/Wizard.jsx

```diff
@@ -0,0 +1,109 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { Row, Button, Col, Nav, NavItem } from 'react-bootstrap';
+
+import WizardStyle from './Wizard.css';
+
+/**
+ * Component that renders a wizard, letting the consumers of the component
+ * manage the state. It will render at least two columns: First column will hold
+ * the steps the wizard will take. Second column will render the component of the
+ * selected step. In a optional third column the consumer can render a preview.
+ */
+class Wizard extends React.Component {
+  static propTypes = {
+    /**
+     * Array of objects which will describe the wizard. The object must
+     * contain a unique 'key' attribute, a 'title' which will be shown as step link on the left side and
+     * a 'component' attribute which will hold the component which is to render for the step.
+     *
+     * e.g:
+     * [\[<br />
+     * &nbsp;&nbsp;{key: 'key1', title: 'General Information', component: (&lt;Acomponent1 /&gt;)},<br />
+     * &nbsp;&nbsp;{key: 'key2', title: 'Details', component: (&lt;Acomponent2 /&gt;)},<br />
+     * &nbsp;&nbsp;{key: 'key3', title: 'Preview', component: (&lt;Acomponent3 /&gt;)},<br />
+     * \]]
+     */
+    steps: PropTypes.arrayOf(PropTypes.object).isRequired,
+    /**
+     * Callback which is called when the user changes the step. As an argument the callback gets the key
+     * of the next step.
+     */
+    onStepChange: PropTypes.func,
+    /** Optional component which can be rendered on the right side e.g a preview */
+    children: PropTypes.element,
+  };
+
+  static defaultProps = {
+    children: undefined,
+    onStepChange: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedStep: props.steps[0].key,
+    };
+
+    this._wizardChanged = this._wizardChanged.bind(this);
+    this._onNext = this._onNext.bind(this);
+    this._onPrevious = this._onPrevious.bind(this);
+  }
+
+  _wizardChanged(eventKey) {
+    this.props.onStepChange(eventKey);
+    this.setState({ selectedStep: eventKey });
+  }
+
+  _disableButton(direction) {
+    const len = this.props.steps.length;
+    const position = direction === 'next' ? (len - 1) : 0;
+    return this.props.steps[position].key === this.state.selectedStep;
```

Comment:
Sometimes wizards need the user to complete the current step before going any further. I guess that's something we will need to take care of sooner or later, but I'm okay if we wait until we really need it.